src/Components/Carousel/Carousel.js

```diff
@@ -25,7 +38,19 @@ const Carousel = () => {
         
     },[])
 
-    console.log(width)
+    useEffect(() => {
+        if( offset === 0) {
+            setActivePage(0)
+        } else if ( offset > 0 && offset < numberReq - count){
+            setActivePage(1)
+        } else if (offset + count == numberReq) {
+            setActivePage(2)
+        }
+    })
```

Comment:
useEffect contains a call to 'setActivePage'. Without a list of dependencies, this can lead to an infinite chain of updates. To fix this, pass [offset, numberReq, count] as a second argument to the useEffect Hook.